### PR TITLE
Fix: Update Table Component's HTML Parser To Not Break When Encountering HTML Comments

### PR DESCRIPTION
**Pull request by sghoweri**
## Jira
http://vjira2:8080/browse/BDS-1665

## Summary
Fixes the `<bolt-table>` web component JavaScript to better handle HTML comments in the initial HTML when initially rendering.

## Details
HTML comments are a different node type (different from text nodes + normal DOM nodes) which wasn't included in the original parser logic for this.

image

## Broken Schema Table (Left) vs Fixed (Right)
<p>
  
  
</p>

## How to test 
- [ ] Review code changes 
- [ ] Confirm new Jest tests added (which include checking the component rendering when HTML comments are in the original content) are passing

### CODE REVIEW

**Comment by remydenton** [comment]
@sghoweri , could you add steps to reproduce the original issue?  I don't see it on https://boltdesignsystem.com/pattern-lab/?p=viewall-components-button, which I believe is the page your screenshot is from.  I also tried adding an HTML comment to a table cell locally and nothing broke.

Also, https://fix-handle-html-comments-in-table-compone.boltdesignsystem.com/pattern-lab/?p=viewall-components-button does not show vertical lines between cell as in your "Fixed" screenshot.  Should it?

**Comment by remydenton** [comment]
From @sghoweri on Slack (for the benefit of other reviewers):
> the easiest way to replicate the issue is to pull down the latest code on master, wrap a <bolt-table> around the schema props table: https://github.com/bolt-design-system/bolt/blob/master/docs-site/src/components/pattern-lab-utils/schema-docs.twig#L110 to see the error

**Comment by remydenton** [comment]
I didn't get any of my questions answered-- but I guess if @danielamorse is happy then this is "Ready to Merge" ¯\_(ツ)_/¯

File: packages/components/bolt-table/src/table.js [hunk 1]
```diff
@@ -38,13 +38,26 @@ class BoltTable extends withLitHtml() {
...
+        return node.content.length;
+      }
+    });
+  }
+
+  removeCommentNodes(nodes) {
     return nodes.filter(node => {
       if (node.type === 'element') {
-        node.children = this.removeEmptyNodes(node.children);
+        node.children = this.removeCommentNodes(node.children);
```
Comment (sghoweri) [inline packages/components/bolt-table/src/table.js]: @remydenton to answer your question on `removeCommentNodes` vs `removeEmptyNodes` (better now than never!):

I _think_ this fix works by first only returning Element nodes + their children (ie. stripping out comments early on) BEFORE the remaining nodes are processed (removing empty nodes, etc).

I wouldn't be surprised if this logic could get further consolidated — I'd just argue that we should prioritize the fix first (and then refactor / rewrite) after the fire cools down a little 🙂 .

**Comment by sghoweri** [comment]
PR was released with v2.8.4-canary.bde01a443.0

**Comment by sghoweri** [comment]
PR was released with v2.9.0

**Pull request merged**